### PR TITLE
feat: add name history endpoint

### PR DESCRIPTION
**Pull request by jyeshe**
refs #1202

### CODE REVIEW

**Comment by jyeshe** [comment]
Missing some unit tests update...

File: lib/ae_mdw/db/sync/name.ex [hunk 1]
```diff
@@ -143,31 +143,38 @@ defmodule AeMdw.Db.Sync.Name do
...
 
   @spec expire_name(state(), Blocks.height(), Names.plain_name()) :: state()
   def expire_name(state, height, plain_name) do
-    Model.name(expire: expiration) = m_name = State.fetch!(state, Model.ActiveName, plain_name)
+    Model.name(active: active, expire: expiration) =
+      m_name = State.fetch!(state, Model.ActiveName, plain_name)
 
-    deactivate_name(state, height, expiration, m_name, :names_expired)
+    state
+    |> State.put(Model.NameExpired, Model.name_expired(index: {plain_name, active, {nil, -1}}))
```
[the remark this answers is not in the document]
Comment (jyeshe) [reply lib/ae_mdw/db/sync/name.ex:176]: This is for the history of the name which can expire multiple times so we really need to store it :smile: A recursive previous field is not appropriate for the same reason that justified the nested resources refactoring.

File: lib/ae_mdw/names.ex [hunk 2]
```diff
@@ -775,6 +868,30 @@ defmodule AeMdw.Names do
...
   end
 
+  defp render_nested_resource(state, {{_active, {txi, _idx} = txi_idx}, table}) do
+    {inner_type, tx_mod} =
+      case table do
+        Model.NameClaim -> {:name_claim_tx, :aens_claim_tx}
+        Model.NameRevoke -> {:name_revoke_tx, :aens_revoke_tx}
+        Model.NameUpdate -> {:name_update_tx, :aens_update_tx}
+        Model.NameTransfer -> {:name_transfer_tx, :aens_transfer_tx}
+      end
```
[the remark this answers is not in the document]
Comment (jyeshe) [reply lib/ae_mdw/names.ex:878]: The regular expiration doesn't have a transaction. The `NameExpire` was handled in another way.

File: lib/ae_mdw/names.ex [hunk 3]
```diff
@@ -120,6 +122,29 @@ defmodule AeMdw.Names do
...
+      {:ok,
+       {
+         serialize_history_cursor(plain_name, prev_cursor),
+         render_history(state, history_keys),
+         serialize_history_cursor(plain_name, next_cursor)
+       }}
+    rescue
+      e in ErrInput ->
+        {:error, e.message}
+    end
```
[the remark this answers is not in the document]
Comment (jyeshe) [reply lib/ae_mdw/names.ex:145]: The other functions are ignoring invalid cursors but I can use tagged errors instead for sure.

**Review by sborrazas — APPROVED** [review]

**Pull request merged**